### PR TITLE
makemkv: 1.14.0 -> 1.14.1

### PR DESCRIPTION
**Pull request by neilmayhew**
###### Motivation for this change

The tarball for 1.14.0 has gone

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build makemkv

It seems old sources are moved to:
```shell
http://www.makemkv.com/download/old/makemkv-bin-${version}.tar.gz
http://www.makemkv.com/download/old/makemkv-oss-${version}.tar.gz
```
I wonder if we should set those as fallback URLs to avoid a failure at every upstream bump

**Comment by c0bw3b** [comment]
Woops it's unfree >;<

**Comment by Mic92** [comment]
Yes, a mirror would help in future.

File: pkgs/applications/video/makemkv/default.nix [hunk 1]
```diff
@@ -4,17 +4,17 @@
 
 stdenv.mkDerivation rec {
   name = "makemkv-${ver}";
-  ver = "1.14.0";
+  ver = "1.14.1";
   builder = ./builder.sh;
 
   src_bin = fetchurl {
     url = "http://www.makemkv.com/download/makemkv-bin-${ver}.tar.gz";
```
Comment (c0bw3b) [inline pkgs/applications/video/makemkv/default.nix]:
```suggestion
    urls = [
      "http://www.makemkv.com/download/makemkv-bin-${ver}.tar.gz"
      "http://www.makemkv.com/download/old/makemkv-bin-${ver}.tar.gz"
    ];
```

File: pkgs/applications/video/makemkv/default.nix [hunk 2]
```diff
@@ -4,17 +4,17 @@
...
-    sha256 = "1xm5pww6jf3m704y7d7nc2ni2a6ygxwb2c665agg2i059sppwz1f";
+    sha256 = "1n4gjb1531gkvnjzipw63v3zdxmrq5nai9nn6m2ix3lskksjrrhp";
   };
 
   src_oss = fetchurl {
     url = "http://www.makemkv.com/download/makemkv-oss-${ver}.tar.gz";
```
Comment (c0bw3b) [inline pkgs/applications/video/makemkv/default.nix]:
```suggestion
    urls = [
      "http://www.makemkv.com/download/makemkv-oss-${ver}.tar.gz"
      "http://www.makemkv.com/download/old/makemkv-oss-${ver}.tar.gz"
    ];
```

**Review by c0bw3b — CHANGES_REQUESTED** [review]
I just tested rebuilding v1.14.0 with the suggested URL changes and it succeeded (404 on the first url then fetchurl tries the second one).
It will prevent failure on the next release.

**Comment by c0bw3b** [comment]
Closing after #50696 got merged

**Pull request closed without merging**